kernel/src/checkpoint/tests.rs

```diff
@@ -0,0 +1,368 @@
+use std::{path::PathBuf, sync::Arc, time::Duration};
+
+use crate::actions::{Add, Metadata, Protocol, Remove};
+use crate::arrow::array::{ArrayRef, StructArray};
+use crate::arrow::datatypes::{DataType, Schema};
+use crate::checkpoint::{deleted_file_retention_timestamp_with_time, CheckpointWriter};
+use crate::engine::arrow_data::ArrowEngineData;
+use crate::engine::{
+    default::{executor::tokio::TokioBackgroundExecutor, DefaultEngine},
+    sync::SyncEngine,
+};
+use crate::snapshot::Snapshot;
+use crate::utils::test_utils::Action;
+use crate::DeltaResult;
+use crate::Engine;
+use crate::Table;
+
+use arrow_53::{
+    array::{create_array, RecordBatch},
+    datatypes::Field,
+};
+
+use object_store::{memory::InMemory, path::Path, ObjectStore};
+use test_utils::delta_path_for_version;
+use url::Url;
+
+#[test]
+fn test_deleted_file_retention_timestamp() -> DeltaResult<()> {
+    let now = Duration::from_secs(1000).as_millis() as i64;
+
+    // Test cases
+    let test_cases = [
+        // Default case (7 days)
+        (None, now - (7 * 24 * 60 * 60 * 1000)),
+        // Zero retention
+        (Some(Duration::from_secs(0)), now),
+        // Custom retention (2000 seconds)
+        // This results in a negative timestamp which is valid - as it just means that
+        // the retention window extends to before UNIX epoch.
+        (Some(Duration::from_secs(2000)), now - (2000 * 1000)),
+    ];
+
+    for (retention, expected) in test_cases {
+        let result =
+            deleted_file_retention_timestamp_with_time(retention, Duration::from_secs(1000))?;
```

Comment:
So this test is pretending that "now" is 1000 seconds [after what]?

tiny nit:
```suggestion
        let now = Duration::from_secs(1000);
        let result = deleted_file_retention_timestamp_with_time(retention, now)?;
```

Reply:
Good question — in this test, now is a simulated timestamp representing 1000 seconds since the UNIX epoch It doesn’t matter in practice since the test just fixes now at 1000s past UNIX epoch to keep the math predictable. The logic under test only cares about relative time (i.e., now minus retention)

Reply:
After a second look, I cleaned up the test case quite a lot